Source/CronosPlayUnreal/Private/DefiWalletCoreActor.cpp

```diff
@@ -1086,6 +1331,91 @@ void ADefiWalletCoreActor::Erc721TransferFrom(
   }
 }
 
+void ADefiWalletCoreActor::Erc721SafeTransferFrom(
+    FString contractAddress, int walletindex, FString fromAddress,
+    FString toAddress, FString tokenid, FCronosTransactionReceiptRaw &result,
+    bool &success, FString &output_message) {
+  try {
+    if (NULL == _coreWallet) {
+      success = false;
+      output_message = TEXT("Invalid Wallet");
+      return;
+    }
+
+    char hdpath[100];
+    snprintf(hdpath, sizeof(hdpath), "m/44'/%d'/0'/0/%d", EthCoinType,
+             walletindex);
+    rust::cxxbridge1::Box<PrivateKey> privatekey = _coreWallet->get_key(hdpath);
+
+    std::string mycontractaddress = TCHAR_TO_UTF8(*contractAddress);
+    std::string myfromaddress = TCHAR_TO_UTF8(*fromAddress);
+    std::string mytoaddress = TCHAR_TO_UTF8(*toAddress);
+    std::string mycronosrpc = TCHAR_TO_UTF8(*myCronosRpc);
+    std::string mytokenid = TCHAR_TO_UTF8(*tokenid);
+
+    Erc721 erc721 =
+        new_erc721(mycontractaddress, mycronosrpc, myCronosChainID).legacy();
+    ::org::defi_wallet_core::CronosTransactionReceiptRaw receipt =
+        erc721.safe_transfer_from(myfromaddress, mytoaddress, mytokenid,
+                                  *privatekey);
+    convertCronosTXReceipt(receipt, result);
+    success = true;
+
+  } catch (const rust::cxxbridge1::Error &e) {
+    success = false;
+    output_message = FString::Printf(
+        TEXT("CronosPlayUnreal Erc721SafeTransferFrom Error: %s"),
+        UTF8_TO_TCHAR(e.what()));
+  }
+}
+
+void ADefiWalletCoreActor::Erc721SafeTransferFromWithData(
+    FString contractAddress, int walletindex, FString fromAddress,
+    FString toAddress, FString tokenid, TArray<uint8> additionaldata,
+    FCronosTransactionReceiptRaw &result, bool &success,
+    FString &output_message) {
+  try {
+    if (NULL == _coreWallet) {
+      success = false;
+      output_message = TEXT("Invalid Wallet");
+      return;
+    }
+
+    char hdpath[100];
+    snprintf(hdpath, sizeof(hdpath), "m/44'/%d'/0'/0/%d", EthCoinType,
+             walletindex);
+    rust::cxxbridge1::Box<PrivateKey> privatekey = _coreWallet->get_key(hdpath);
```

Comment:
could this be generic over the wallet -- i.e. not hardcode the development wallet, but work for any wallet method (WalletConnect, Web3Auth, Magic/Fortmatic...)??

Reply:
yes, possible, 
https://github.com/crypto-com/defi-wallet-core-rs/issues/453
i'll apply generic signing for (walletconnect, various wallets etc)